docs/api_reference/src/quantities.tex

```diff
@@ -146,121 +464,1387 @@
+consteval auto @\exposidnc{expr-map}@(T);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\exposidnc{expr-map} maps contents of one expression template to another resulting in a different type list.
+
+\pnum
+Let \tcode{Proj} be projection used for mapping, and
+let \tcode{T} be the expression template to map from.
+
+\pnum
+\effects
+TBD.
+
+\pnum
+\returns
+TBD.
+\end{itemdescr}
+
+\rSec1[qty.dim]{Dimension}
+
+\rSec2[qty.dim.general]{General}
+
+\pnum
+Subclause \ref{qty.dim} specifies the components
+for defining the dimension of a quantity\irefiev{112-01-11}.
+
+\rSec2[qty.dim.concepts]{Concepts}
+
+\begin{itemdecl}
+template<typename T>
+concept @\deflibconcept{Dimension}@ = @\exposconceptnc{tag-type}@<T> && std::@\stdconcept{derived_from}@<T, @\exposidnc{dimension-interface}@>;
+
+template<typename T>
+concept @\defexposconceptnc{BaseDimension}@ = @\libconcept{Dimension}@<T> && std::@\stdconcept{derived_from}@<T, base_dimension>;
+
+consteval bool @\exposidnc{is-dimension-one}@(std::meta::info type_alias) {
+  return dealias(type_alias) == ^dimension_one;
+}
+
+template<typename T>
+concept @\defexposconceptnc{IsPowerOfDim}@ =
+  (@\exposidnc{is-specialization-of}@(^T, ^power) &&
+   (@\exposconceptnc{BaseDimension}@<typename T::factor> || @\exposidnc{is-dimension-one}@(^typename T::factor)));
+
+template<typename T>
+constexpr bool @\exposidnc{is-per-of-dims}@ = false;
+
+template<typename... Ts>
+constexpr bool @\exposidnc{is-per-of-dims}@<per<Ts...>> =
+  (... && (@\exposconceptnc{BaseDimension}@<Ts> || @\exposidnc{is-dimension-one}@(^Ts) || @\exposconceptnc{IsPowerOfDim}@<Ts>));
+
+template<typename T>
+concept @\defexposconceptnc{DerivedDimensionExpr}@ =
+  @\exposconceptnc{BaseDimension}@<T> || @\exposidnc{is-dimension-one}@(^T) || @\exposconceptnc{IsPowerOfDim}@<T> || @\exposidnc{is-per-of-dims}@<T>;
+
+template<auto D1, auto D2>
+concept @\defexposconceptnc{SameDimension}@ = @\libconcept{Dimension}@<decltype(D1)> && @\libconcept{Dimension}@<decltype(D2)> && D1 == D2;
+
+template<typename T, auto D>
+concept @\deflibconcept{DimensionOf}@ = @\libconcept{Dimension}@<T> && @\libconcept{Dimension}@<decltype(D)> && @\exposconceptnc{SameDimension}@<T{}, D>;
+\end{itemdecl}
+
+\rSec2[qty.dim.types]{Types}
+
+\begin{codeblock}
+namespace mp_units {
+
+struct @\exposidnc{dimension-interface}@ {
+  friend consteval @\libconcept{Dimension}@ auto operator*(@\libconcept{Dimension}@ auto lhs, @\libconcept{Dimension}@ auto rhs)
+  {
+    return @\exposidnc{expr-multiply}@<derived_dimension, struct dimension_one>(lhs, rhs);
+  }
+
+  friend consteval @\libconcept{Dimension}@ auto operator/(@\libconcept{Dimension}@ auto lhs, @\libconcept{Dimension}@ auto rhs)
+  {
+    return @\exposidnc{expr-divide}@<derived_dimension, struct dimension_one>(lhs, rhs);
+  }
+
+  friend consteval bool operator==(@\libconcept{Dimension}@ auto lhs, @\libconcept{Dimension}@ auto rhs)
+  {
+    return ^decltype(lhs) == ^decltype(rhs);
```

Comment:
I'd hope so.

Reply:
The specification is really only about behavior and not optimizing compile times -- for sure nothing in the standard is currently specified in terms of reflection -- because of course reflection itself isn't in the standard.